presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,513 @@
+import com.esri.core.geometry.OperatorOverlaps;
+import com.esri.core.geometry.OperatorSimpleRelation;
+import com.esri.core.geometry.OperatorTouches;
+import com.esri.core.geometry.OperatorWithin;
+import com.esri.core.geometry.Point;
+import com.esri.core.geometry.Polygon;
+import com.esri.core.geometry.SpatialReference;
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.esri.core.geometry.ogc.OGCLineString;
+import com.esri.core.geometry.ogc.OGCMultiLineString;
+import com.esri.core.geometry.ogc.OGCPoint;
+import com.esri.core.geometry.ogc.OGCPolygon;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import org.apache.commons.lang.StringUtils;
+
+import java.lang.reflect.Method;
+
+import static com.facebook.presto.geospatial.GeometryUtils.deserialize;
+import static com.facebook.presto.geospatial.GeometryUtils.serialize;
+import static java.util.Arrays.asList;
+
+public final class GeoFunctions
+{
+    private static final String LINE = "LineString";
+    private static final String MULTILINE = "MultiLineString";
+    private static final String MULTIPOINT = "MultiPoint";
+    private static final String POLYGON = "Polygon";
+    private static final String POINT = "Point";
+    private static final String GEOMETRY = "Geometry";
+
+    private GeoFunctions() {}
+
+    @Description("Construct Line Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stLineString(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.fromText(input.toStringUtf8()), "st_line_string", LINE);
+        return serialize(geometry);
+    }
+
+    @Description("Construct Point Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stPoint(@SqlType(StandardTypes.DOUBLE) double longitude, @SqlType(StandardTypes.DOUBLE) double latitude)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.createFromEsriGeometry(new Point(longitude, latitude), null), "st_point", POINT);
+        return serialize(geometry);
+    }
+
+    @Description("Construct Polygon Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stPolygon(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(OGCGeometry.fromText(input.toStringUtf8()), "st_polygon", POLYGON);
+        return serialize(geometry);
+    }
+
+    @Description("Returns the area of the geometry if it is a polygon or multi-polygon.")
+    @ScalarFunction("st_area")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stArea(@SqlType(GEOMETRY) Slice input)
+    {
+        return deserialize(input).getEsriGeometry().calculateArea2D();
```

Comment:
Implementation of deserialize(input) allow for null return value; if deserialize returns null, this line will throw NPE. Is this intentional? Would it be easier to troubleshoot if deserialize throws exception on corrupted input?